packages/vite/src/node/server/ws.ts

```diff
@@ -30,10 +35,32 @@ export function createWebSocketServer(
       }
     })
   } else {
+    const websocketServerOptions: WebSocket.ServerOptions = {}
+    const port = hmr ? hmr.port : 24678
```

Comment:
**nitpick:** This could even more simplified with modern syntax :slightly_smiling_face: 

```suggestion
    const port = hmr?.port ?? 24678
```

Reply:
You go back to the original problem here, `?.` cannot be used on a union containing things other than an object:

<img width="825" alt="image" src="https://user-images.githubusercontent.com/10807310/116282446-0fa9a180-a78b-11eb-8e21-022dec57de15.png">

Reply:
Damnd, I fall into this trap so much, cause I just read the code without an IDE in the GitHub Web UI :confused: